llvm/test/ExecutionEngine/JITLink/AArch32/ELF_static_arm_reloc.s

```diff
@@ -50,6 +50,36 @@ jump24_target:
 	bx	lr
 	.size	jump24_target,	.-jump24_target
 
+
+# CHECK-TYPE: {{[0-9a-f]+}} R_ARM_MOVW_ABS_NC data_symbol
+# CHECK-INSTR: 	00000010 <movw>:
+# CHECK-INSTR: 	      10: e3000000     movw      r0, #0x0
+# jitlink-check: decode_operand(movw, 1) = (data_symbol&0x0000ffff)
+	.globl	movw
+	.type	movw,%function
+	.p2align	2
+movw:
+	movw r0, :lower16:data_symbol
+	.size	movw,	.-movw
+
+# CHECK-TYPE: {{[0-9a-f]+}} R_ARM_MOVT_ABS data_symbol
+# CHECK-INSTR: 	00000014 <movt>:
+# CHECK-INSTR: 	      14: e3400000     movt      r0, #0x0
+# jitlink-check: decode_operand(movt, 2) = (data_symbol&0xffff0000>>16)
```

Comment:
I assume we have to check the operand at index 2 here, because the disassembler adds an implicit predicate operand? That was the case with the `BL` instruction:
https://github.com/llvm/llvm-project/blob/b6f66c94bce7d9c0ef029d133b85925ba23198b4/llvm/test/ExecutionEngine/JITLink/AArch32/ELF_static_thumb_reloc.s#L16-L26

Might be worth a comment.

Reply:
Exactly, the `BL` would result the following MC operands:
```
<MCInst #4386 tBL <MCOperand Imm:14> <MCOperand Reg:0> <MCOperand Imm:0> <MCOperand Reg:3>>
```
Here in `MOVT` we have:
```
<MCInst #932 MOVTi16 <MCOperand Reg:73> <MCOperand Reg:73> <MCOperand Imm:30463> <MCOperand Imm:14> <MCOperand Reg:0>>
```
Will add the comment to clarify.

Reply:
Small nit: "movt generates **one** leading implicit predicate **operand**" -- it has an explicit register operand at index 0